dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/MetadataBitstreamRestRepository.java

```diff
@@ -464,10 +501,46 @@ public String extractFile(InputStream inputStream, String fileType) {
         if (fileCounter > maxPreviewCount) {
             sb.append("<element>...too many files...|0</element>");
         }
-        sb.append(("</root>"));
+        sb.append("</root>");
         return sb.toString();
     }
 
+    /**
+     * Extracts files from an InputStream, processes them based on the specified file type (tar or zip),
+     * and returns an XML representation of the file paths.
+     * @param inputStream the InputStream containing the file data
+     * @param fileType    the type of file to extract ("tar" or "zip")
+     * @return an XML string representing the extracted file paths
+     */
+    public String extractFile(InputStream inputStream, String fileType) {
+        List<String> filePaths = new ArrayList<>();
+        Path tempFile = null;
+        FileSystem zipFileSystem = null;
+
+        try {
+            // Create a temporary file based on the file type
+            tempFile = createTempFile(fileType);
+
+            // Copy the input stream to the temporary file
+            Files.copy(inputStream, tempFile, StandardCopyOption.REPLACE_EXISTING);
+
+            // Process the file based on its type
+            if ("tar".equals(fileType)) {
```

Comment:
What am I seeing? You have defined a constant for this and still using a raw string? nonono